From 8fa1f7da877180258f18e47d131f76af37d4464c Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Thu, 30 Apr 2020 14:09:30 +0800
Subject: [PATCH] Remove uid/gid check for device property files

Basing on the need of rootless process, device property files
maybe created by a non-root process, this uid/gid check is
improper for this scenario.

Change-Id: Ie24b5fd0ff05caa81a40982a7fe409e61dac8946
Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 libc/system_properties/prop_area.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/libc/system_properties/prop_area.cpp b/libc/system_properties/prop_area.cpp
index 42bee9f3e..c0ab5af39 100644
--- a/libc/system_properties/prop_area.cpp
+++ b/libc/system_properties/prop_area.cpp
@@ -110,8 +110,7 @@ prop_area* prop_area::map_fd_ro(const int fd) {
     return nullptr;
   }
 
-  if ((fd_stat.st_uid != 0) || (fd_stat.st_gid != 0) ||
-      ((fd_stat.st_mode & (S_IWGRP | S_IWOTH)) != 0) ||
+  if (((fd_stat.st_mode & (S_IWGRP | S_IWOTH)) != 0) ||
       (fd_stat.st_size < static_cast<off_t>(sizeof(prop_area)))) {
     return nullptr;
   }
-- 
2.20.1

